### DISCUSSION TITLE
Exercise 2: Show that the union of C1 and C2 is consistent.

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#27] [Exercises]
**Exercise 2:** Let C1 and C2 be two consistent cuts. Show that the union of C1 and C2 is consistent.

**Comment by rotiroti** [comment]
We aim to prove the consistency of the union, $C_1 \cup C_2$, and this requires us to show that:

$$(e' \in C_1 \cup C_2) \land (e \to e') \Rightarrow e \in C_1 \cup C_2.$$

Let's break down the proof step by step:

- $(e' \in C_1 \cup C_2) \land (e \to e')$, as per the precondition for consistency.
- $(e' \in C_1 \lor e' \in C_2)$, by the definition of set union.
- Consider two cases separately:
    - If $e' \in C_1$, and since $C_1$ is consistent, $e \in C_1$.
    - If $e' \not \in C_1 \land C_2$, then $e' \in C_2$, and since $C_2$ is consistent, $e \in C_2$.
- $(e \in C_1 \cup C_2)$, by the definition of set union.

**Comment by rotiroti** [comment]
The previous solutions contain some errors. Following is an updated version of it.

We aim to prove the consistency of the union, $C_1 \cup C_2$, and this requires us to show that:

$$(e' \in C_1 \cup C_2) \land (e \to e') \Rightarrow e \in C_1 \cup C_2.$$

Let's break down the proof step by step:

- $(e' \in C_1 \cup C_2) \land (e \to e')$, as per the precondition for consistency.
- $(e' \in C_1 \lor e' \in C_2) \land (e \to e')$, by the definition of set union.
- $(e' \in C_1 \land e \to e') \lor (e' \in C_2 \land e \to e')$, since $C_1$ and $C_2$ are consistent cuts.
- This implies that either $(e \in C_1) \lor (e \in C_2)$ or both.
- By the definition of set union, we can conclude that $e \in C_1 \cup C_2$.

**Comment by cornflexxx** [comment]
Suppose $C_1 \cup C_2$ is inconsistent then there exists $e\not\in C_1\cup C_2$ s.t. $e' \in C_1 \cup C_2 \land e\to e'$, but if $e$ is not in the union  it cannot be in either $C_1$ or $C_2$, so one or both cuts are inconsistent.
